lib/cpp/src/thrift/protocol/TProtocol.h

```diff
@@ -659,6 +659,11 @@ class TProtocolFactory {
   virtual ~TProtocolFactory() {}
 
   virtual boost::shared_ptr<TProtocol> getProtocol(boost::shared_ptr<TTransport> trans) = 0;
+  virtual boost::shared_ptr<TProtocol> getProtocol(boost::shared_ptr<TTransport> inTrans,
```

Comment:
This makes sense.  I would like that information captured in a comment somewhere.  Maybe in THeaderTransport.h?